src/CMakeLists.txt

```diff
@@ -292,7 +292,18 @@ IF (LIBSSH2_FOUND)
 	LIST(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
 	#SET(LIBGIT2_PC_LIBS "${LIBGIT2_PC_LIBS} ${LIBSSH2_LDFLAGS}")
 
-	CHECK_LIBRARY_EXISTS("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+	# We might need to provide additional libraries so libssh2 links
+	SET(CMAKE_REQUIRED_LIBRARIES_OLD ${CMAKE_REQUIRED_LIBRARIES})
```

Comment:
`CMAKE_REQUIRED_LIBRARIES` is only used for CMake's internals such that it can successfully link test executables, right? If so, I'd expect this dance around `CMAKE_REQUIRED_LIBRARIES_OLD` to be useless. Just set `CMAKE_REQUIRED_LIBRARIES` and unset it afterwards again. Or does it have a default value?

Reply:
Looks like it is there for that express purpose, yes. I'll just unconditionally unset.